sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/NumberPatternShouldBeRegular.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        private int field = 1_000;
+
+        public void CompliantCases()
+        {
+            var thousand = 1000;
+
+            // decimal notation
+            var balance = 2_435_951.68;
```

Comment:
You could have `_` after the decimal point, like `111_2.22_333` - what should be the correct behavior in such cases? I feel that we should threat the two sides around the decimal point as different numbers, but still have the same group length... Please, add some tests.